query.go

```diff
@@ -353,21 +381,21 @@ func (q *query) queryPeer(ch chan<- *queryUpdate, p peer.ID) {
 
 	// dial the peer
 	if err := q.dht.dialPeer(dialCtx, p); err != nil {
-		ch <- &queryUpdate{unreachable: []peer.ID{p}}
+		ch <- &queryUpdate{cause: p, unreachable: []peer.ID{p}}
 		return
 	}
 
 	// add the peer to the global set of queried peers since the dial was successful
 	// so that no other disjoint query tries sending an RPC to the same peer
 	if !q.globallyQueriedPeers.TryAdd(p) {
-		ch <- &queryUpdate{unreachable: []peer.ID{p}}
+		ch <- &queryUpdate{cause: p, unreachable: []peer.ID{p}}
 		return
 	}
 
 	// send query RPC to the remote peer
 	newPeers, err := q.queryFn(queryCtx, p)
 	if err != nil {
-		ch <- &queryUpdate{unreachable: []peer.ID{p}}
+		ch <- &queryUpdate{cause: p, unreachable: []peer.ID{p}}
 		return
 	}
 
```

Comment:
@petar @aschmahmann 

Don't we want to differentiate between a peer being unreachable because of a dial or query failure for the LookupEvent ? It would be good to have this for tracing because a successful dial but a failed query implies we dialled a non DHT peer which means something has wrong with a peer's Routing Table.